lightning/restore/restore.go

```diff
@@ -892,7 +892,7 @@ func (t *TableRestore) restoreEngines(ctx context.Context, rc *RestoreController
 		for engineID, engine := range cp.Engines {
 			select {
 			case <-ctx.Done():
-				return ctx.Err()
```

Comment:
without breaking the rest of the code may still run after a Ctrl+C

Reply:
yes, the rest code will still run for a while to shut down gracefully. After setting the engineErr, this for loop will break at line #898~900, then the WaitGroup will wait for all the sub-tasks to be finished before return an error.

This is essential because if we return here, the outer logic will close the pebble db which may cause the running sub-tasks panic by reading or writing to a closed db.

Reply:
can you put that explanation into the code comment?

Reply:
done